From 43c68262a6d5b59289779f6ab1160b438f5666f9 Mon Sep 17 00:00:00 2001
From: Nikita Yaschenko <Nafanya@users.noreply.github.com>
Date: Fri, 10 Jan 2020 22:46:47 +0000
Subject: [PATCH 16/24] Fix strtod.cc undefined constants (#123)

When DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS is not defined
there is a build error when -Wall -Werror enabled because
kMaxExactDoubleIntegerDecimalDigits, exact_powers_of_ten and
kExactPowersOfTenSize are used only in else branch of this define
(when it's defined).

Fixes https://github.com/google/double-conversion/issues/122
---
 double-conversion/strtod.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/double-conversion/strtod.cc b/double-conversion/strtod.cc
index d0bb7f7..3da7a42 100644
--- a/double-conversion/strtod.cc
+++ b/double-conversion/strtod.cc
@@ -35,10 +35,12 @@
 
 namespace double_conversion {
 
+#if defined(DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS)
 // 2^53 = 9007199254740992.
 // Any integer with at most 15 decimal digits will hence fit into a double
 // (which has a 53bit significand) without loss of precision.
 static const int kMaxExactDoubleIntegerDecimalDigits = 15;
+#endif // #if defined(DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS)
 // 2^64 = 18446744073709551616 > 10^19
 static const int kMaxUint64DecimalDigits = 19;
 
@@ -55,6 +57,7 @@ static const int kMinDecimalPower = -324;
 static const uint64_t kMaxUint64 = DOUBLE_CONVERSION_UINT64_2PART_C(0xFFFFFFFF, FFFFFFFF);
 
 
+#if defined(DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS)
 static const double exact_powers_of_ten[] = {
   1.0,  // 10^0
   10.0,
@@ -82,6 +85,7 @@ static const double exact_powers_of_ten[] = {
   10000000000000000000000.0
 };
 static const int kExactPowersOfTenSize = DOUBLE_CONVERSION_ARRAY_SIZE(exact_powers_of_ten);
+#endif // #if defined(DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS)
 
 // Maximum number of significant digits in the decimal representation.
 // In fact the value is 772 (see conversions.cc), but to give us some margin
-- 
2.29.2

